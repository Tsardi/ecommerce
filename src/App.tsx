import React from 'react';
import Home from './components/Home';
import Cart from './components/Cart';
import './App.css';

const App: React.FC = () => {
  return (
    <div className="App">
      <header className="App-header">
        <h1>E-Commerce Store</h1>
      </header>
      <main>
        <Home />
        <Cart />
      </main>
    </div>
  );
};

export default App;