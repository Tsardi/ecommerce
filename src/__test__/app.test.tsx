import React from 'react';
import { render, screen } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { Provider } from 'react-redux';
import store from '../app/store'; // Correct import!
import App from '../App';

jest.mock('axios');

describe('App', () => {
  test('renders the main heading', () => {  //does the app render the main heading
    const queryClient = new QueryClient(); // fetching and caching updating asynchronous data in react
    render(
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          <App />
        </QueryClientProvider>
      </Provider>
    );
    expect(screen.getByText(/E-Commerce Store/i)).toBeInTheDocument();
  });
});
test('matches the snapshot', () => { //does the app match the snapshot
  const queryClient = new QueryClient();
  const { asFragment } = render(
    <Provider store={store}>
      <QueryClientProvider client={queryClient}>
        <App />
      </QueryClientProvider>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});