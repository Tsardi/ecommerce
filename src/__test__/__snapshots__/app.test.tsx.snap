// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <h1>
        E-Commerce Store
      </h1>
    </header>
    <main>
      <div>
        Loading...
      </div>
      <div
        class="cart"
      >
        <h2>
          Shopping Cart
        </h2>
        <p>
          Your cart is empty.
        </p>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
